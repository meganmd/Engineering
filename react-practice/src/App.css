.App {
  text-align: center;
}

.logInButton{
}

table {
  margin-top: 350px;
    border-collapse: collapse;
    width: 35%;
}

.loginForm{
  margin-top: 50px;
}

#card
{
  background: #000000;
  color: black;
}

#productbacklog
{
  left:.3%;
}

#inprogress
{
  left:60.3%;
}

#todo
{
  left:40.3%;
}

#done
{
  left:80.3%;
}

#scrumbacklog
{
  left:20.3%;
}

#title
{
  background-color:#000000;
  text-align:center;
  color:white;
  padding: 2px 0px;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
}

#todo, #inprogress, #done, #productbacklog, #scrumbacklog
{
  position:absolute;
  width: 19.6%;
  min-height:75%;
  border: 2px solid black;
  outline: 0;
  background: #f2f2f2;
  box-sizing: border-box;
  border-radius: 15px;
}

th, td {
    padding: 8px;
    text-align: center;
    border-bottom: 1px solid #ddd;
}

tr:hover {background-color: #EEEEEE}

button:hover {background-color: #61DAFB;
color: #000000}

.textStuff{
    font-family: "Roboto", sans-serif;
}

button {
  font-family: "Roboto", sans-serif;
  text-transform: uppercase;
  outline: 0;
  background: #000000;
  width: 14%;
  border: 0;
  margin: 15px;
  padding: 15px;
  color: #FFFFFF;
  font-size: 14px;
  -webkit-transition: all 0.3 ease;
  transition: all 0.3 ease;
  cursor: pointer;
}

input {
  font-family: "Roboto", sans-serif;
  outline: 0;
  background: #f2f2f2;
  width: 30%;
  border: 0;
  margin: 0 0 15px;
  padding: 15px;
  box-sizing: border-box;
  font-size: 14px;
}

textarea{
  font-family: "Roboto", sans-serif;
  outline: 0;
  background: #f2f2f2;
  width: 30%;
  border: 0;
  margin: 0 0 15px;
  padding: 15px;
  box-sizing: border-box;
  font-size: 14px;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 200px;
  padding: 20px;
  color: white;
}

.App-header-black {
  background-color: #222;
  height: 200px;
  padding: 20px;
  color: white;
}

.App-header-red {
  background-color: #DC143C;
  height: 200px;
  padding: 20px;
  color: white;
}

.App-header-blue {
  background-color: #0000FF;
  height: 200px;
  padding: 20px;
  color: white;
}

.App-header-green {
  background-color: #228B22;
  height: 200px;
  padding: 20px;
  color: white;
}

.App-header-purple {
  background-color: #8B008B;
  height: 200px;
  padding: 20px;
  color: white;
}

.App-header-orange {
  background-color: #FF4500;
  height: 200px;
  padding: 20px;
  color: white;
}

.App-header-beige {
  background-color: #F5F5DC;
  height: 200px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
