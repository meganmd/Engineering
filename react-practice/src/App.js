import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import LogInForm from './LogInForm';
import UserTable from './UserTable';
import CreateProjectForm from './CreateProjectForm';
import AddUserForm from './AddUserForm';
import Client from './Client';
import s from './index.css';
import AddUserToProjectForm from './AddUserToProjectForm'
import ProjectTable from './ProjectTable.js'
import PBIBacklogForm from './PBIBacklogForm'
import CreatePBIForm from './CreatePBIForm'
import InvitedProjectsTableForm from './InvitedProjectsTableForm'

function LogOutButton(props) {
  return (
    <button onClick={props.onClick}>
      Logout
    </button>
  );
}

function CreateProjectButton(props) {
  return(
      <button onClick={props.createClick}>
        Create Project
      </button>
  );
}


function addUser(username,password,firstName,lastName){
  Client.addUser(username,password,firstName, lastName, function(){});
}

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {isLoggedIn: false, loggedInUser: {}, isCreatingProject: false, currentProject: {},isViewingProject:false};
    this.handleLogIn = this.handleLogIn.bind(this);
    this.handleLogOut = this.handleLogOut.bind(this);
    this.handleCreateProject = this.handleCreateProject.bind(this);
    this.handleProjectComplete = this.handleProjectComplete.bind(this);
    this.handleLeaveCreateProjectForm = this.handleLeaveCreateProjectForm.bind(this);
    this.handleProjectSelected = this.handleProjectSelected.bind(this);
    this.handleLeavePBIBacklogForm = this.handleLeavePBIBacklogForm.bind(this);
  }

  handleLogIn(user){
    this.setState({isLoggedIn: true, loggedInUser: user});
  }

  handleLogOut(){
    this.setState({isLoggedIn: false, loggedInUser: ''});
  }

  handleCreateProject(){
    this.setState({isCreatingProject: true});
  }

  handleProjectComplete(){
    console.log('arriving here');
    this.setState({isCreatingProject: false});
  }

  handleLeaveCreateProjectForm(){
    this.setState({isCreatingProject: false});
  }

  handleProjectSelected(project){
    this.setState({currentProject:project, isViewingProject:true});
  }

  handleLeavePBIBacklogForm(){
    this.setState({currentProject: {}, isViewingProject:false});
  }

  render() {
    var greeting = null;
    var content = [];

    if(this.state.isLoggedIn){
      if(this.state.isCreatingProject){
        greeting = 'Create Project Form';
        content = <CreateProjectForm user={this.state.loggedInUser}
          handleProjectComplete={this.handleProjectComplete}
          handleLeaveCreateProjectForm={this.handleLeaveCreateProjectForm}/>
      } else if(this.state.isViewingProject){
        content.push( <PBIBacklogForm project={this.state.currentProject}
          handleLeavePBIBacklogForm={this.handleLeavePBIBacklogForm} />);
      }else{
        var top = <h1  id="projectTitleText">
                      <LogOutButton onClick={this.handleLogOut} />
                      {this.state.loggedInUser.firstName + " " + this.state.loggedInUser.lastName}
                     <CreateProjectButton createClick={this.handleCreateProject}/>
                   </h1>
        content.push(top);
        var projectTable = <div id="projectHome">
                              <ProjectTable user={this.state.loggedInUser} handleProjectSelected={this.handleProjectSelected}/>
                          </div>
        content.push(projectTable);
        var invitedTable = <div id="invitedProjects">
          <InvitedProjectsTableForm user={this.state.loggedInUser} />
        </div>
        content.push(invitedTable);
      }
    } else {
      greeting = 'Please Login...';
      content = <LogInForm getUser={Client.getUser} addUser={addUser} handleLogIn={this.handleLogIn} handleLogOut={this.handleLogOut} isLoggedIn={this.state.isLoggedIn}/>;
    }
    return (
      <div className="App">
        {content}
      </div>
    );
  }
}
/*
{content}
*/
//
export default App;
