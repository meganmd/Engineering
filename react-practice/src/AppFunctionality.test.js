import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
//import Link from '../Link.react';
import renderer from 'react-test-renderer';

it('renders correctly (incomplete)', () => {
  const tree = renderer.create(
    <App />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

//Log-in log-out possibility
//this is more for testing out the app, should eventually be given its
//test file.
/*
it('can set logged in', () => {
    App.handleLogin({isLoggedIn: true, loggedInUser: 'mockName'});
    expect(Client1.getUser('mockName', cb).isLoggedIn.toBeTruthy());
});

it('can set logged out', () => {
    App.handleLogOut({isLoggedIn: false, loggedInUser: 'mockName'});
    expect(this.state.isLoggedIn.toBeFalsy());
});
*/
