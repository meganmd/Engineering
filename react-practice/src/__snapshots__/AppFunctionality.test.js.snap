exports[`test renders correctly (incomplete) 1`] = `
<div
  className="App">
  <div
    className="App-header-black">
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg" />
    <h1>
      Scrumtious Demo Page
    </h1>
    <h2>
      Please Login...
    </h2>
  </div>
  <br />
  <div
    className="loginForm">
    <div
      className="loginForm">
      <input
        name="userfield"
        onChange={[Function]}
        placeholder="Enter Username..."
        type="text" />
      <br />
      <input
        name="passwordfield"
        onChange={[Function]}
        placeholder="Enter Password..."
        type="text" />
       
      <br />
      <button
        className="logInButton"
        onClick={[Function]}>
        Login
      </button>
      <button
        className="switchToRegisteringButton"
        onClick={[Function]}>
        Register
      </button>
      <br />
      <font
        color="red">
        
      </font>
    </div>
  </div>
  <div
    className="PBIBacklogDisplay">
    <div
      id="Backlog">
      <h1>
        Test Project
      </h1>
      <div
        id="board">
        <div
          id="productbacklog"
          style={
            Object {
              "background": "#f2f2f2",
              "border": "2px solid black",
              "border-radius": "10px",
              "box-sizing": "border-box",
              "height": 525,
              "outline": "0",
              "position": "absolute",
              "width": "19.6%",
            }
          }>
          <div
            id="title">
            Product Backlog
          </div>
          <div
            style={
              Object {
                "background": "#dfdfdf",
                "border": "2px solid black",
                "border-radius": "15px",
                "left": "5%",
                "min-height": "100px",
                "position": "absolute",
                "top": 35,
                "width": "90%",
              }
            }>
            <br />
            This is a user story
            <br />
            Size: 
            SMALL
          </div>
          <div
            style={
              Object {
                "background": "#dfdfdf",
                "border": "2px solid black",
                "border-radius": "15px",
                "left": "5%",
                "min-height": "100px",
                "position": "absolute",
                "top": 160,
                "width": "90%",
              }
            }>
            <br />
            Make Food
            <br />
            Size: 
            LARGE
          </div>
          <div
            style={
              Object {
                "background": "#dfdfdf",
                "border": "2px solid black",
                "border-radius": "15px",
                "left": "5%",
                "min-height": "100px",
                "position": "absolute",
                "top": 285,
                "width": "90%",
              }
            }>
            <br />
            This is a user story
            <br />
            Size: 
            SMALL
          </div>
          <div
            style={
              Object {
                "background": "#dfdfdf",
                "border": "2px solid black",
                "border-radius": "15px",
                "left": "5%",
                "min-height": "100px",
                "position": "absolute",
                "top": 410,
                "width": "90%",
              }
            }>
            <br />
            Make Food
            <br />
            Size: 
            LARGE
          </div>
        </div>
        <div
          id="scrumbacklog"
          style={
            Object {
              "background": "#f2f2f2",
              "border": "2px solid black",
              "border-radius": "10px",
              "box-sizing": "border-box",
              "height": 525,
              "outline": "0",
              "position": "absolute",
              "width": "19.6%",
            }
          }>
          <div
            id="title">
            Scrum Backlog
          </div>
          <div
            style={
              Object {
                "background": "#dfdfdf",
                "border": "2px solid black",
                "border-radius": "15px",
                "left": "5%",
                "min-height": "100px",
                "position": "absolute",
                "top": 35,
                "width": "90%",
              }
            }>
            <br />
            Do some stuff
            <br />
            Size: 
            SMALL
          </div>
          <div
            style={
              Object {
                "background": "#dfdfdf",
                "border": "2px solid black",
                "border-radius": "15px",
                "left": "5%",
                "min-height": "100px",
                "position": "absolute",
                "top": 160,
                "width": "90%",
              }
            }>
            <br />
            Another user story
            <br />
            Size: 
            LARGE
          </div>
          <div
            style={
              Object {
                "background": "#dfdfdf",
                "border": "2px solid black",
                "border-radius": "15px",
                "left": "5%",
                "min-height": "100px",
                "position": "absolute",
                "top": 285,
                "width": "90%",
              }
            }>
            <br />
            Another user story
            <br />
            Size: 
            LARGE
          </div>
        </div>
        <div
          id="todo"
          style={
            Object {
              "background": "#f2f2f2",
              "border": "2px solid black",
              "border-radius": "10px",
              "box-sizing": "border-box",
              "height": 525,
              "outline": "0",
              "position": "absolute",
              "width": "19.6%",
            }
          }>
          <div
            id="title">
            To Do
          </div>
          <div
            style={
              Object {
                "background": "#dfdfdf",
                "border": "2px solid black",
                "border-radius": "15px",
                "left": "5%",
                "min-height": "100px",
                "position": "absolute",
                "top": 35,
                "width": "90%",
              }
            }>
            <br />
            Stop being lazy
            <br />
            Size: 
            SMALL
          </div>
          <div
            style={
              Object {
                "background": "#dfdfdf",
                "border": "2px solid black",
                "border-radius": "15px",
                "left": "5%",
                "min-height": "100px",
                "position": "absolute",
                "top": 160,
                "width": "90%",
              }
            }>
            <br />
            Another user story
            <br />
            Size: 
            LARGE
          </div>
        </div>
        <div
          id="inprogress"
          style={
            Object {
              "background": "#f2f2f2",
              "border": "2px solid black",
              "border-radius": "10px",
              "box-sizing": "border-box",
              "height": 525,
              "outline": "0",
              "position": "absolute",
              "width": "19.6%",
            }
          }>
          <div
            id="title">
            In Progress
          </div>
          <div
            style={
              Object {
                "background": "#dfdfdf",
                "border": "2px solid black",
                "border-radius": "15px",
                "left": "5%",
                "min-height": "100px",
                "position": "absolute",
                "top": 35,
                "width": "90%",
              }
            }>
            <br />
            Finish the project
            <br />
            Size: 
            SMALL
          </div>
          <div
            style={
              Object {
                "background": "#dfdfdf",
                "border": "2px solid black",
                "border-radius": "15px",
                "left": "5%",
                "min-height": "100px",
                "position": "absolute",
                "top": 160,
                "width": "90%",
              }
            }>
            <br />
            Another user story
            <br />
            Size: 
            LARGE
          </div>
        </div>
        <div
          id="done"
          style={
            Object {
              "background": "#f2f2f2",
              "border": "2px solid black",
              "border-radius": "10px",
              "box-sizing": "border-box",
              "height": 525,
              "outline": "0",
              "position": "absolute",
              "width": "19.6%",
            }
          }>
          <div
            id="title">
            Done
          </div>
          <div
            style={
              Object {
                "background": "#dfdfdf",
                "border": "2px solid black",
                "border-radius": "15px",
                "left": "5%",
                "min-height": "100px",
                "position": "absolute",
                "top": 35,
                "width": "90%",
              }
            }>
            <br />
            This is another user story
            <br />
            Size: 
            SMALL
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
